src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -166,12 +166,27 @@ public async Task ExecuteAsync(ExportJobRecord exportJobRecord, WeakETag weakETa
                 // but we will not be updating the job record.
                 _logger.LogTrace("The job was updated by another process.");
             }
+            catch (SecretStoreException sse)
+            {
+                _logger.LogError(sse, "Secret store error. The job will be marked as failed.");
+
+                _exportJobRecord.FailureReason = sse.Message;
+                await CompleteJobAsync(OperationStatus.Failed, cancellationToken);
+            }
+            catch (DestinationConnectionException dce)
+            {
+                _logger.LogError(dce, "Can't connect to destination. The job will be marked as failed.");
+
+                _exportJobRecord.FailureReason = dce.Message;
+                await CompleteJobAsync(OperationStatus.Failed, cancellationToken);
+            }
             catch (Exception ex)
             {
                 // The job has encountered an error it cannot recover from.
                 // Try to update the job to failed state.
                 _logger.LogError(ex, "Encountered an unhandled exception. The job will be marked as failed.");
 
+                _exportJobRecord.FailureReason = "Unknown Error";
```

Comment:
Can this be ex.Message? If so, then these 3 conditions could just be collapsed into this one and you can output the error message in the log?

Reply:
Since we will be returning data in `FailureReason` to the end-user, I want to avoid setting it to `ex.Message` because we can't know for certain what it's value will be in cases of unhandled exceptions.